netplan/cli/commands/set.py

```diff
@@ -71,15 +73,55 @@ def command_set(self):
         with tempfile.TemporaryFile() as tmp:
             libnetplan.create_yaml_patch(yaml_path, value, tmp)
             tmp.flush()
+
             tmp.seek(0, io.SEEK_SET)
             parser.load_nullable_fields(tmp)
+            if filename:
+                # Validate the full YAML hierarchy without ignoring any NetDefs.
+                # Exit early if this YAML patch would lead to an invalid config.
+                # XXX: The origin file of each individual YAML setting/stanza
+                #      should be tracked individually, to avoid this
+                #      double-parsing workaround (LP: #2003727)
+                try:
+                    validation_parser = libnetplan.Parser()
+                    tmp.seek(0, io.SEEK_SET)
+                    validation_parser.load_nullable_fields(tmp)
+                    validation_parser.load_yaml_hierarchy(self.root_dir)
+                    tmp.seek(0, io.SEEK_SET)
+                    validation_parser.load_yaml(tmp)
+                    validation_state = libnetplan.State()
+                    validation_state.import_parser_results(validation_parser)
+                except libnetplan.LibNetplanException as e:
+                    logging.error('New settings invalidate existing configuration: ' + str(e))
```

Comment:
nitpick (not-blocking): we're actually handling the validation error in a cleaner way in this very specific case than in the general case, where we basically just let import_parser_results throw an exception and hope something will catch it up the stack.

Reply:
I will keep it this way, as the reason (override) for this specific parsing error might not be clear to the user, so I think it makes sense to give some extra context here.